backend/webserver/views.py

```diff
@@ -70,6 +72,109 @@ def post(self, request):
+            if author.id == request.user.id:
+                post.edited_at = datetime.datetime.utcnow().replace(tzinfo=utc)
+                serializer = UpdatePostSerializer(instance=post,data=request.data,partial=True,context={'request': request})
+                if serializer.is_valid():
+                    serializer.save()
+                    return Response(serializer.data, status=status.HTTP_200_OK)
+                return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+            else:
+                return Response({'message': 'You cannot edit another authors post'}, status=status.HTTP_400_BAD_REQUEST) 
+        else:
+            return Response({'message': 'You can only edit public posts'}, status=status.HTTP_400_BAD_REQUEST)
+        
+
+    def delete(self,request,pk,post_id,*args,**kwargs):
+        author =self.get_author(pk=pk)
+        post = self.get_post(post_id,author.id)
+        if post.visibility == "PUBLIC":
+            if author.id == request.user.id:
+                post.delete()
+                return Response({"message":"Object deleted!"}, status=status.HTTP_200_OK)
+            else:
+                return Response({'message': 'You cannot delete another authors post'}, status=status.HTTP_400_BAD_REQUEST) 
+        return Response({'message': 'You can only delete public posts'}, status=status.HTTP_400_BAD_REQUEST)
+
+
+class AllPosts(APIView):
+    authentication_classes = [BasicAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+    def get_author(self,pk):
+        author = get_object_or_404(Author,pk=pk)
+        return author
+
+    def get(self, request,pk, *args, **kwargs):
+        author = self.get_author(pk)
+        posts = author.post_set.all().order_by("-created_at")
+        serializer = PostSerializer(posts, many=True, context={'request': request})
+        return Response(serializer.data, status=status.HTTP_200_OK)
+    
+    
+    def post(self, request, pk, *args, **kwargs):
+        author = self.get_author(pk)
+        serializer = CreatePostSerializer(data=request.data)
+        if author.id == request.user.id:
+            if serializer.is_valid():
+                new_post = Post.objects.create(
+                    author=author,
+                    title=serializer.data["title"],
+                    description=serializer.data["description"],
+                    unlisted=serializer.data["unlisted"],
+                    content_type=serializer.data["content_type"],
+                    content=serializer.data["content"],
+                    visibility=serializer.data["visibility"]
+                )
+                if new_post.visibility == "FRIENDS" or new_post.visibility == "PUBLIC":
+                    for follow in author.followed_by_authors.iterator():
+                        with transaction.atomic():
+                            Inbox.objects.create(target_author=follow.follower,post=new_post)
```

Comment:
With this line, you're also sharing the a PUBLIC post with only the followers in of the target author. This is _incorrect_. Public post should be shared in inbox either all authors on the server or no one at all (depending on the answer you get on your forum question).